### ISSUE TITLE
How to generate/modify the default profile?

### ISSUE THREAD

**Issue by AliAskari** [conan-io/conan#1541]
Before version `0.25` the `conan.conf` file contained the default settings of conan. `conan.conf` was generated on the very first call to `conan`. So simply calling `conan` would generate the default settings.

In version `0.25`/`0.25.1` the default settings from conan.conf have been removed. The idea is to have different profiles under `~/.conan/profiles/` and use them according to the build environment/settings. I think [this](https://github.com/conan-io/conan/issues/1374) covers the major points.

The problem is that the default profile, `~/.conan/profiles/default` is not generated unless I call `conan install`. On the other hand calling `conan install` without any arguments or paths to a recipe will fail and returns a non zero error code. 

1. I would like to know why the default profile is not generated during the very first call to `conan`?
2. Is there a way to generate the default profile without getting a non zero error code from conan? During building of our docker/vagrant images I am failing the build process if any of the commands fail. This is causing all of docker/vagrant build pipelines to fail.

I used to create pseudo "profiles" before each build by calling `conan config set arch/build_type/...`.  I assume since you have moved to profiles you don't use the settings in `conan.conf` file any more.

3. Is there a way to create a profile with a similar command? I know the profiles are simple text files but I prefer to create the profiles using a conan command rather than creating a text file under the profiles folder. Something like `conan profile create custom_profile -s arch=x86 -s build_type=Debug` and any unspecified option is inherited from the default profile.

Thanks

**Comment by memsharded** [comment]
Sorry for breaking this. When we moved the [settings_defaults] to a default profile, we did the migration to not break, but we didn't consider the usage of ``conan config`` to set the default profile, because we mostly use it to ``config`` conan config variables, but not the [settings_defaults], cause the most recommended approach was already to use profiles.

So if you were using ``conan config`` to do it, it won't be possible directly with conan 0.25.

How are you calling ``conan``? Shell scripts? CI scripts? Don't you have some way to capture the error code? Currently there is no way that ``conan install`` will not fail without a conanfile. You might install some existing package, with ``conan install Pkg/version@user/channel`` but that would be slower and unnecessary.

I think the best workaround at the moment would be running something like:

```bash
$ printf "[settings]\narch=x86\nbuild_type=Debug\n...." > /home/user/.conan/profiles/default
```

We will be adding the equivalent to ``conan config`` for profiles asap (0.26). We have to think about the UX, I am not sure that the default behavior taking unspecified options from the default would be the easiest to understand, as there should be an option to create totally new profiles, independent of the default. Maybe another different subcommand. I will post here any progress. Thanks very much for reporting this issue.

**Comment by AliAskari** [comment]
@memsharded I see, thanks for the explanation. I have been using Conan for more than 9 months now and at the time I started writing our scripts I was not aware of the profiles.

I have already modified our CI scripts to handle this error but it is not really clean the way I want it. I am currently writing the profile text files in our CI. It would be really good if there was new functionality in Conan so that in case of a change in the the format of the profile files we wouldn't need to re modify the scripts. Is there a time line for the releases?

**Comment by memsharded** [comment]
@AliAskari At the time you wrote the scripts, I think the profiles didn't exist. We added them later, while we learned from our users, and became evident that sharing configuration in text files should be a must

The idea of the profiles files is that they shouldn't break if we add or change something. We could be very wrong in something and have to do some breaking change, but not likely, profiles seems relatively stable feature now, and even if in beta, we always try not to break. This time, your use case escaped from our analysis :(

We are releasing approx every 4 weeks. Next release should be expected by the 20th of August.

**Comment by lasote** [comment]
Let's purpose a syntax for create and configure a profile before implementing it:

**$ conan profile new <name_or_abs_path>**  ==> Creates a profile with empty sections
$ conan profile new /path/to/mynewprofile

**$ conan profile update <section_name> <string_value> <profile_name_or_path>**
$ conan profile update settings "compiler.libcxx=libstdc++" MyProfile => Smart to override libcxx
$ conan profile update env MYVAR=23 /path/to/profile => Smart to override MYVAR value or add if not exists
$ conan profile update build_requires `mylib/1.0@lasote/testing` /path/to/profile => Won't add if duplicated

Not sure about the command name:

**$ conan profile delete_value <section_name> <string_value> <profile_name_or_path>**
$ conan profile delete_value env MYVAR=23 /path/to/profile

**Comment by memsharded** [comment]
```bash
$ conan profile update settings.compiler.libcxx="libstdc++"  myprofile# Value should strip quote, and accept without them
$ conan profile remove settings.compiler.libcxx myprofile
```
not sure if profile name/path should be first or second

**Comment by lasote** [comment]
Now for a build_require.
About the conan profile remove command... that means remove a profile.

**Comment by memsharded** [comment]
Well it means remove a profile if not specified a certain setting/value. If you specify a setting value, it can only remove that setting/value.

```bash
$ conan profile update build_requires="MyPkg: cmake/version@user/channel"
```

**Comment by lasote** [comment]
> $ conan profile update build_requires="MyPkg: cmake/version@user/channel"

It looks like the build_requires should be exactly `MyPkg: cmake/version@user/channel`, not an append.

About the remove, if you read in the help "conan profile remove" subcommand, what would you think about what the command does?

**Comment by lasote** [comment]
will be available in the next release.

**Comment by memsharded** [comment]
The ``new``, ``update``, ``remove`` subcommands have been added to ``conan profile``, released in 0.26, please update and try them.

**Comment by liberforce** [comment]
As the option is not mentionned here: `conan profile new foo --detect` will create a filled profile named `foo`.
